src/fparser/two/Fortran2003.py

```diff
@@ -6952,26 +7019,39 @@ def match(string):
             spec = splitted[idx].strip()
             spec = repmap(spec)
             if idx == 0 and "=" not in spec:
-                # Must be a unit number. However, we do not prepend "UNIT="
-                # to it in case the following Io_Control_Spec is positional
-                # (and therefore either a Format or Namelist spec).
+                # Must be a unit number (C910). However, we do not prepend
+                # "UNIT=" to it in case the following Io_Control_Spec is
+                # positional (and therefore either a Format (C917) or Namelist
+                # spec (C918)).
                 lst.append(Io_Control_Spec(spec))
                 unit_is_positional = True
-            elif idx == 1 and "=" not in spec:
-                if not unit_is_positional:
-                    # Cannot have a positional argument following a
-                    # named argument
-                    return
-                # Without knowing the type of the variable named in spec
-                # we have no way of knowing whether this is a format or
-                # a namelist specifier. However, if it is a character
-                # constant or "*" then it must be a Format spec and we can
-                # prepend "FMT=" to it.
-                spec = spec.lstrip().rstrip()
+            elif idx == 1:
+                spec = spec.strip()
                 if Char_Literal_Constant.match(spec) or \
                    StringBase.match("*", spec):
+                    # This second argument is a character constant or "*". It
```

Comment:
It would make sense, however, e.g. `Format.match('"(I3)"')` returns `False`. R914 for Format says:

    R914 format is default-char-expr
                   or label
                   or *

and `default-char-expr` is just an `expr` although, looking at the spec, there is Note 9.27 that says "A default-char-expr includes a character constant."  Currently, we have:

    class Default_Char_Expr(Base):  # R726
        """
        <default-char-expr> = <expr>
        """
        subclass_names = ['Expr']

which does not include a character constant. The logical solution then is to extend its `subclass_names` to include `Char_Literal_Constant`.

Reply:
However, if I do that we then match any variable name which highlights the fact that we have no way of disambiguating a name-list name from a format string held in a variable.

Reply:
Actually, that's OK so I've left it as is.